### DISCUSSION TITLE
Shortcode

### DISCUSSION THREAD

**Discussion by matthieu2607** [Cecilapp/Cecil#1571] [Q&A] [answered]
Hi @ArnaudLigny,

I looked at the documentation to create a "shortcode" but I can't get it to work on my project.

I created a 'shortcodes.twig' file in the root of the 'layouts' folder with the following code:

```twig
{% extends 'macros.twig' %}

{% block macros %}

    {% macro don(label = 'Faire un don en ligne') %}
        <div class="text-center">
            <a href="" class="btn" target="_blank">{{ label }}</a>
        </div>
    {% endmacro %}

{% endblock %}
```

When I then want to generate it on a markdown file, nothing happens : 

```markdown

My text...

{{ shortcode.don() }}

My text...

```

Thanks in advance for your help ;)

Best regards,

Matthieu

**Comment by matthieu2607** [comment]
After checking, the youtube shortcode also doesn't work. I must be missing something in the configuration.

**Comment by ArnaudLigny** [comment, marked as answer]
Hello @matthieu2607,

Thanks for your report: there is effectively a bug in this experimental feature, sorry :-/

I fixed it in the last version of Cecil, and updated the documentation (https://cecil.app/documentation/content/#shortcodes).

Important point: you must use `{% extends 'extended/macros.twig' %}` instead of `{% extends 'macros.twig' %}`.

You can see an example on my website: https://arnaudligny.fr/test/shortcodes.

Source files:
- template: https://github.com/ArnaudLigny/arnaudligny.fr/blob/master/layouts/test/shortcodes.html.twig
- custom shortcodes: https://github.com/ArnaudLigny/arnaudligny.fr/blob/master/layouts/shortcodes.twig
- page: https://github.com/ArnaudLigny/arnaudligny.fr/blob/master/pages/test/shortcodes.md

**Comment by matthieu2607** [comment]
Big thank @ArnaudLigny 

I updated Cecil to 7.25.5 but still got the problem.

The repository is here: https://github.com/Pixel-Developpement/defiducoeur

Thanks

**Reply by ArnaudLigny** [reply]
Hello @matthieu2607,

I thinks this is related to the specific way to include the page content when you wants to use the Shortcode feature.

You must use `{{ include(page.content_template) }}` (instead of `{{ page.content }}`) in your template (it seams it should be the case for https://github.com/Pixel-Developpement/defiducoeur/blob/main/layouts/programme.html.twig).

Refer to the _important_ note in documentation: https://cecil.app/documentation/content/#dynamic-content.

**Comment by matthieu2607** [comment]
Hi @ArnaudLigny 

I'm an idiot, I didn't apply the code "{{ include(page.content_template) }}" on the right template :(

Thanks, it works ;)

This functionality of Cecil is top I think

Big thanks
